src/endpoint/s3/s3_utils.js

```diff
@@ -668,6 +672,109 @@ function response_field_encoder_url(value) {
     return new URLSearchParams({ 'a': value }).toString().slice(2); // slice the leading 'a='
 }
 
+class S3SelectStream extends Transform {
```

Comment:
It's not a small class, and this s3_utils module is big enough, so I would prefer to move it to it's own file, the natural pick would be `src/util/s3select.js`